# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/WebRenderTypes.h
# Commit: 11451cccc88c
# Full Hash: 11451cccc88c077e8fe763d4231bbf1af5226061
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-12 04:37:49
# Regressor Bug: 1869346
# File Overlap Count: 5
# Description:
#   Bug 1869346 - Rework how canvas accesses shared surfaces in the compositor process. r=gfx-reviewers,lsalzman
#   
#   This patch adds plumbing to allow sharing ExternalImageId via
#   SurfaceDescriptor. This will be used in a future patch by WebGL and
#   WebGPU to avoid extra copies. It also refactors how CanvasTranslator
# ==============================================================================

diff -r 8043d7693dba -r 11451cccc88c gfx/webrender_bindings/WebRenderTypes.h
--- a/gfx/webrender_bindings/WebRenderTypes.h	Mon Dec 11 18:15:34 2023 +0000
+++ b/gfx/webrender_bindings/WebRenderTypes.h	Mon Dec 11 18:22:21 2023 +0000
@@ -903,6 +903,8 @@
                                 : wr::HasScrollLinkedEffect::No;
 }
 
+enum class ExternalImageSource : uint8_t { Unknown = 0, SharedSurfaces, Last };
+
 }  // namespace wr
 }  // namespace mozilla
 
